contracts/gol.cairo

```diff
@@ -0,0 +1,450 @@
+%lang starknet
+%builtins pedersen range_check bitwise
+
+from starkware.cairo.common.alloc import alloc
+from starkware.cairo.common.cairo_builtins import (HashBuiltin,
+    BitwiseBuiltin)
+from starkware.cairo.common.math import (assert_not_zero,
+    assert_le_felt, assert_not_equal, unsigned_div_rem, split_felt)
+from starkware.starknet.common.syscalls import get_caller_address
+from starkware.cairo.common.memcpy import memcpy
+from starkware.cairo.common.uint256 import Uint256
+
+from contracts.utils.life_rules import evaluate_rounds
+from contracts.utils.packing import (unpack_game, pack_game)
+from contracts.utils.constants import (DIM, CREATE_CREDIT_REQUIREMENT,
+    INFINITE_GAME_GENESIS, GIVE_LIFE_CREDIT_REQUIREMENT)
+
+from openzeppelin.upgrades.library import Proxy
+from openzeppelin.token.erc20.library import ERC20
```

Comment:
Does it include all views & externals?

Reply:
afaik, only the functions within the namespace

Reply:
So we need to expose ERC20 functions.